### ISSUE TITLE
OutOfMemoryException when form is empty

### ISSUE THREAD

**Issue by EccelorRavi** [Http-Multipart-Data-Parser/Http-Multipart-Data-Parser#33]
Hi Jake,

Tried posting an empty form. The constructor of parser takes on some time and then throws System.OutOfMemoryException. Can you replicate the issue and verify?

**Comment by Vodurden** [comment]
Hi RaviVadera,

Thanks for reporting this issue and sorry for the late reply. I haven't had a chance to check this out yet but I just wanted to let you know it's on my todo list.

I'll let you know when I've solved the issue. Otherwise feel free to submit a pull request :+1: 

Regards,

**Comment by BaileyDan** [comment]
Hi RaviVadera,

Figured i'd try and help out and take a look at this issue.

Can you provide a sample of the data you are submitting, I am trying to replicate here and the worst I am getting is an unexpected end of stream exception.

Cheers

Dan.

**Comment by RaviVadera** [comment]
Hi, Dan.

I appreciate your help. I am submitting an empty form without any parts.

**Comment by Jericho** [comment]
I know you reported this issue a long time ago (more than 4 years!) and appreciate your patience. I recently took over stewardship of this project and I'm trying to address outstanding issues.

I have been investigating the problem you reported and, while I haven't experienced any `OutOfMemory` exceptions, I do observe that the parser in unable to handle empty forms. 

So far, I found two problems related to empty forms

1. When the `MultipartFormDataParser.Parse` is invoked and the boundary parameter is omitted, the parser attempts to figure out the boundary that separates each parameter but there are no such boundary in the form. The attempt to figure out the boundary fails with a  `end of stream` exception that @BaileyDan mentioned.

2. When the `MultipartFormDataParser.Parse` is invoked and the boundary parameter is specified, the parser assumes that there is a boundary to mark the first parameter but, again, there are no parameters and therefore it fails to locate the boundary. This leads to a `Could not find expected boundary` exception.

The second issue I identified is actually trivial and I already have a solution for it but the first issue is more complex and I will need some time to find a solution. I'll keep you updated as I make progress.
